java/vector/src/main/codegen/templates/ComplexCopier.java

```diff
@@ -63,6 +64,10 @@ private static void writeValue(FieldReader reader, FieldWriter writer) {
           writer.start();
           for(String name : reader){
             FieldReader childReader = reader.reader(name);
+            if (writer instanceof UnionMapWriter) {
```

Comment:
can you add a comment explaing the logic here?

Reply:
Complex copier didn't call UnionMapWriter startEntry/key/value API(which call setAddVectorAsNullable), thus the nullable info was not set properly, we should set these infos from input FieldReader (key vector set non-nullable, value vector set nullable)